cassandra-analytics-common/src/main/java/org/apache/cassandra/spark/reader/EmptyStreamScanner.java

```diff
@@ -24,13 +24,13 @@ public class EmptyStreamScanner implements StreamScanner<Rid>
     public static final EmptyStreamScanner INSTANCE = new EmptyStreamScanner();
 
     @Override
-    public Rid rid()
+    public Rid data()
```

Comment:
Since you are renaming the method, can you rename the class to make it easier to understand? I'd propose `RowData`, instead of `Rid`

Is the method name `data()` clear enough to indicate that it returns a single element? In this code snippet `scanner.data()`, could it be read as all the data it has scanned? If so, it is clearer to call it `nextData`. 

I'd agree with @aweisberg that we should just move to the more common and well established interface, `Iterator`.

Reply:
I suggest you take a deeper look at `AbstractStreamScanner` and the work involved in refactoring to an `java.util.Iterator`, `hasNext()` has the expectation of being idempotent and this is not currently the case for `AbstractStreamScanner`. I am +1 on moving to `java.util.Iterator` but it should be handled as an independent PR so it can be reviewed carefully and tested outside of these changes.

Reply:
Sounds good.